test/Dialect/SV/prettify-verilog.mlir

```diff
@@ -240,3 +241,325 @@ hw.module @unary_sink_no_duplicate(%arg0: i4) -> (result: i4) {
+  // CHECK:   [[FIELD_OUTER:%.+]] = sv.array_index_inout [[FIELD_INNER]][%c1_i2_0] : !hw.inout<array<3xi2>>, i2
+  // CHECK:   sv.passign [[FIELD_OUTER:%.+]], %value : i2
+  // CHECK: }
+
+  // VERILOG:      always @(posedge clock)
+  // VERILOG-NEXT:   r[2'h1][2'h1] <= value;
+}
+
+// CHECK-LABEL: hw.module private @ConnectNestedFieldsAndIndices
+hw.module private @ConnectNestedFieldsAndIndices(%clock: i1, %reset: i1, %value: i2) -> () {
+  %c2_i3 = hw.constant 2 : i3
+  %c0_i3 = hw.constant 0 : i3
+  %c-2_i2 = hw.constant -2 : i2
+  %c0_i2 = hw.constant 0 : i2
+  %c1_i2 = hw.constant 1 : i2
+  %c1_i3 = hw.constant 1 : i3
+
+  %r = sv.reg : !hw.inout<!hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>>
+  %val = sv.read_inout %r : !hw.inout<!hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>>
+  sv.always posedge %clock {
+    sv.passign %r,  %17 : !hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>
+  }
+
+  %c1_i3_0 = hw.constant 1 : i3
+  %5 = hw.array_get %val[%c1_i3_0] : !hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>
+  %6 = hw.struct_extract %5["a"] : !hw.struct<a: !hw.array<3xstruct<b: i2>>>
+  %c1_i2_1 = hw.constant 1 : i2
+  %7 = hw.array_get %6[%c1_i2_1] : !hw.array<3xstruct<b: i2>>
+  %8 = hw.struct_inject %7["b"], %value : !hw.struct<b: i2>
+  %9 = hw.array_slice %6[%c0_i2] : (!hw.array<3xstruct<b: i2>>) -> !hw.array<1xstruct<b: i2>>
+  %10 = hw.array_create %8 : !hw.struct<b: i2>
+  %11 = hw.array_slice %6[%c-2_i2] : (!hw.array<3xstruct<b: i2>>) -> !hw.array<1xstruct<b: i2>>
+  %12 = hw.array_concat %9, %10, %11 : !hw.array<1xstruct<b: i2>>, !hw.array<1xstruct<b: i2>>, !hw.array<1xstruct<b: i2>>
+  %13 = hw.struct_inject %5["a"], %12 : !hw.struct<a: !hw.array<3xstruct<b: i2>>>
+  %14 = hw.array_slice %val[%c0_i3] : (!hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>) -> !hw.array<1xstruct<a: !hw.array<3xstruct<b: i2>>>>
+  %15 = hw.array_create %13 : !hw.struct<a: !hw.array<3xstruct<b: i2>>>
+  %16 = hw.array_slice %val[%c2_i3] : (!hw.array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>) -> !hw.array<3xstruct<a: !hw.array<3xstruct<b: i2>>>>
+  %17 = hw.array_concat %14, %15, %16 : !hw.array<1xstruct<a: !hw.array<3xstruct<b: i2>>>>, !hw.array<1xstruct<a: !hw.array<3xstruct<b: i2>>>>, !hw.array<3xstruct<a: !hw.array<3xstruct<b: i2>>>>
+  // CHECK: sv.always posedge %clock {
+  // CHECK:   [[ARR_1:%.+]] = sv.array_index_inout %r[%c1_i3] : !hw.inout<array<5xstruct<a: !hw.array<3xstruct<b: i2>>>>>, i3
+  // CHECK:   [[STRUCT_A:%.+]] = sv.struct_field_inout [[ARR_1]]["a"] : !hw.inout<struct<a: !hw.array<3xstruct<b: i2>>>>
+  // CHECK:   [[ARR_1:%.+]] = sv.array_index_inout [[STRUCT_A]][%c1_i2] : !hw.inout<array<3xstruct<b: i2>>>, i2
+  // CHECK:   [[STRUCT_B:%.+]] = sv.struct_field_inout [[ARR_1]]["b"] : !hw.inout<struct<b: i2>>
+  // CHECK:   sv.passign [[STRUCT_B]], %value : i2
+  // CHECK: }
+
+  // VERILOG:      always @(posedge clock)
+  // VERILOG-NEXT:   r[3'h1].a[2'h1].b <= value;
+}
```

Comment:
Really appreciate the number of tests that check for the various ways this lowering can go :+1: